tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -1430,27 +1432,92 @@ void CGMSHLSLRuntime::AddHLSLFunctionInfo(Function *F, const FunctionDecl *FD) {
       streamIndex++;
     }
 
+    unsigned GsInputArrayDim = 0;
     if (parmDecl->hasAttr<HLSLTriangleAttr>()) {
-      funcProps->ShaderProps.GS.inputPrimitive = DXIL::InputPrimitive::Triangle;
+      if (funcProps->ShaderProps.GS.inputPrimitive ==
```

Comment:
There is a lot of repeated patterns in this if/else code. Can we make a function to factor out the common code?